### ISSUE TITLE
Documenting `createErrorMessage` and allowing custom message templates?

### ISSUE THREAD

**Issue by lolmaus** [offirgolan/ember-cp-validations#39]
I found assembling custom validator messages to be burdensome. I decided to have a look how it's done in default validators and discovered that a nice helper `createErrorMessage` is used.

I ended up doing this:

```js
import Ember from 'ember';
import BaseValidator from 'ember-cp-validations/validators/base';

// Monkey-patching the `messages` object
import messages from 'bealang-frontend-ember/validators/messages';
messages.noWhitespaceAround = 'should not have whitespace at the beginning or end'

export default BaseValidator.extend({
  validate(value, options, model, attribute) {
    if (value.length === value.trim().length) {
      return true;
    }

    // Using the secret helepr
    return this.createErrorMessage('noWhitespaceAround', options, value);
  }
});
```

Suggestions:

1. `createErrorMessage` should be documented in the readmy and not hidden from the public. ^_^
2. Should be able to extend `messages` gracefully in a way that prevents from accidentally overwriting existing messages (but allows explicit overwriting).

**Comment by offirgolan** [comment]
> 1. `createErrorMessage` should be documented in the readmy and not hidden from the public. ^_^

:+1:

> 2. Should be able to extend `messages` gracefully in a way that prevents from accidentally overwriting existing messages (but allows explicit overwriting).

This should be solved once we move over to our i18n solution #9

**Comment by offirgolan** [comment]
Closing in favor of #40 and #9.
